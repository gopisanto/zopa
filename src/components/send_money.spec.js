import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from '../reducers';
import SendMoney from './send_money';

Enzyme.configure({ adapter: new Adapter() });

let store;

describe('SendMoney form component', () => {
  beforeEach(() => {
    store = createStore(reducer, applyMiddleware(thunk));
  });

  test('should render', () => {
		const wrapper = renderer.create(
      <Provider store={store}>
        <SendMoney />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
