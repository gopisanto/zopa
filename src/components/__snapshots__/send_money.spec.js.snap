// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMoney form component should render 1`] = `
<div
  className="send-money"
>
  <form
    autoComplete="none"
    onSubmit={[Function]}
  >
    <h1>
      Send money
    </h1>
    <div
      className="ti"
    >
      <label
        className="ti-lbl"
        htmlFor={undefined}
      >
        Name
      </label>
      <div>
        <input
          autoComplete="none"
          className="ti-field"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="ti"
    >
      <label
        className="ti-lbl"
        htmlFor={undefined}
      >
        Email address
      </label>
      <div>
        <input
          autoComplete="none"
          className="ti-field"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="ti"
    >
      <label
        className="ti-lbl"
        htmlFor={undefined}
      >
        Amount
      </label>
      <div>
        <span>
          Â£
        </span>
        <input
          autoComplete="none"
          className="ti-field currency"
          name="amount"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="currency"
          value=""
        />
      </div>
    </div>
    <input
      className="btn-send"
      type="submit"
      value="Send"
    />
  </form>
</div>
`;
